misc/printk.c

```diff
@@ -94,6 +99,10 @@ void _vprintk(out_func_t out, void *ctx, const char *fmt, va_list ap)
 	int min_width = -1;
 	int long_ctr = 0;
 
+#ifdef CONFIG_SMP
+	k_spinlock_key_t key = k_spin_lock(&lock);
```

Comment:
printk() runs in user mode. It is not a system call, it's just a library function.
k_spin_lock() is not callable from user mode.
If you enable CONFIG_SMP and CONFIG_USERSPACE on the same board this is going to just crash all threads trying to call printk().